src/probnum/_config.py

```diff
@@ -36,39 +42,58 @@ class Configuration:
     0.01
     """
 
+    def __init__(self) -> None:
+        object.__setattr__(self, "_options_registry", dict())
+
+    def __hasattr__(self, key: str) -> bool:
+        return key in self._options_registry
+
+    def __getattr__(self, key):
+        if not self.__hasattr__(key):
```

Comment:
Another thing I would like to understand, but I don't:
Using `hasattr(self, key)` here does not work. My only explanation is that it depends on the order in which python registers the methods somehow? If somebody knows, I'd appreciate a brief explanation!